test/Serilog.Tests/Context/LogContextTests.cs

```diff
@@ -349,6 +349,24 @@ public void DoesNotThrowOnCrossDomainCallsWhenLeaseExpired()
             void CallFromRemote() { }
         }
 #endif
+
+        static async Task TestWithSyncContext(Func<Task> testAction, SynchronizationContext syncContext)
+        {
+            var prevCtx = SynchronizationContext.Current;
+            SynchronizationContext.SetSynchronizationContext(syncContext);
+
+            Task t;
+            try
+            {
+                t = testAction();
```

Comment:
Not sure at all about what I'm saying, but ... shouldn't there be an `await` here (i.e. in the `try` instead of **after** the `finally` ?

Reply:
Thanks for the feedback! The problem with reverting sync context after await is that after await point I'm not owning thread anymore and this thread will lose its original context  (`Xunit.Sdk.AsyncTestSyncContext`)  so I'm not sure that this will be safe for other tests or xunit. You can set/reset safely while you own the thread, for example, for sync implementations. That's why I can't use here  disposable pattern.